pilot/pkg/xds/monitoring.go

```diff
@@ -82,9 +83,10 @@ var (
 
 	// TODO: Update all the resource stats in separate routine
 	// virtual services, destination rules, gateways, etc.
-	xdsClients = monitoring.NewGauge(
```

Comment:
Why make it a sum? This is not a counter.

Reply:
So I don't have to recompute (or keep track of in the discovery server) the version split every time I have to update a gauge. Which of sum, recompute or track would you prefer?

Reply:
Sum is not correct measure here. What we are interested in is the exact number of proxy client connected at a particular time point, instead of rate or delta of proxy connected between two time points.  I think it makes sense to track it here with a map from proxy version to the corresponding gauge.

Reply:
Wouldnt that still require recomputing each time? or are you suggesting having a gauge but with a value stored alongside it that we can increment and decrement before recording?

Reply:
Yes, that is I am thinking. The map is {proxy_version: (number of connected proxy, gauge.with(version_tag, proxy_version))}.

Reply:
Saw this after, I updated the PR. Let me know if my changes align.